packages/machine-extractor/src/extractMachinesFromFile.ts

```diff
@@ -13,15 +13,17 @@ export const extractMachinesFromFile = (
   }
 
   const { file, machineNodes } = getMachineNodesFromFile(fileContent);
+  const machines = machineNodes.map((node) =>
+    getMachineExtractResult({
+      file,
+      fileContent,
+      node,
+    }),
+  );
 
   return {
-    machines: machineNodes.map((node) =>
-      getMachineExtractResult({
-        file,
-        fileContent,
-        node,
-      }),
-    ),
+    machines,
+    options: machines.map((m) => m?.getAllMachineOptions()),
```

Comment:
Why do we need this here? Can't we just always get *specific* options from a *specific* machine when needed?